packages/graphql-language-service/src/utils/getVariablesJSONSchema.ts

```diff
@@ -106,14 +106,29 @@ const scalarTypesMap: { [key: string]: JSONSchema6TypeName } = {
   DateTime: 'string',
 };
 
+class Marker {
+  private set = new Set<string>()
+  mark(name: string): boolean {
+    if (this.set.has(name)) {
+      return false
+    } else {
+      this.set.add(name)
+      return true
+    }
+  }
+}
+
 /**
  *
  * @param type {GraphQLInputType}
+ * @param options
+ * @param definitionMarker
  * @returns {DefinitionResult}
  */
 function getJSONSchemaFromGraphQLType(
   type: GraphQLInputType | GraphQLInputField,
   options?: JSONSchemaOptions,
+  definitionMarker: Marker = new Marker(),
 ): DefinitionResult {
   let required = false;
```

Comment:
Just one change I'd like to suggest here:

```ts
function getJSONSchemaFromGraphQLType(
  type: GraphQLInputType | GraphQLInputField,
  options?: JSONSchemaOptions,
  definitionMarker: Marker = new Marker(),
): DefinitionResult {
  let required = false;
  ````
  
I think should become:
  
  ```ts
function getJSONSchemaFromGraphQLType(
  type: GraphQLInputType | GraphQLInputField,
  options?: JSONSchemaOptions,
): DefinitionResult {
  const definitionMarker = new Marker();
  let required = false;
  ````
  
  Unless there is a reason we need to expose this argument on this private, internal method?

Reply:
> Putting it into options is indeed a good way, but it seems that this will expose the marker to the upper layer. I plan to encapsulate JSONSchemaOptions as follows.

Ohhh I understand better now, that makes a lot of sense!